staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -386,6 +386,13 @@ func (o *ApplyOptions) Run() error {
 			klog.V(4).Infof("error recording current command: %v", err)
 		}
 
+		if len(info.Name) == 0 {
+			metadata, _ := meta.Accessor(info.Object)
+			generatedName := metadata.GetGenerateName()
+			if len(generatedName) > 0 {
+				return fmt.Errorf("apply using generate name %s not supported", generatedName)
```

Comment:
Could this perhaps be the following instead:

```suggestion
				return fmt.Errorf("cannot use generateName with kubectl apply")
```

My thought is that actually adding the `generatedName` to the error suggests something is wrong with the name itself. I think `cannot use generateName with kubectl apply` or something similar is more clear.

Reply:
Sure. But there's a little concern is that user applies multiple objects and I think the user need to know which object has the problem. By the way, seems word `kubectl` normally won't be in the error message.

Reply:
@zhouya0 Yes, my apologies. I see your point for including `generatedName`.  And yes, if `kubectl` is not normally included, I think just using `apply` makes sense. 

Perhaps something like:

`fmt.Errorf("from %s: cannot use generateName with apply", generatedName)`

WDYT?

Reply:
Great :) . I changed it.

Reply:
Do you think it will be helpful to tell use kubectl create instead of apply and also what are the issues to get this fixed with apply itself?

Reply:
I not sure, just I found there's no where in code suggesting user "not using a kind of command and use another command".